core/sail/fts/lucene/src/test/java/org/eclipse/rdf4j/sail/lucene/AbstractGenericLuceneTest.java

```diff
@@ -0,0 +1,914 @@
+	 * we experienced problems with the NULL context and lucenesail in August 2008
+	 *
+	 * @throws Exception
+	 */
+	@Test
+	public void testNullContextHandling()
+		throws Exception
+	{
+		connection.add(SUBJECT_4, PREDICATE_1, vf.createLiteral("sfourponecone"));
+		connection.add(SUBJECT_4, PREDICATE_2, vf.createLiteral("sfourptwocone"));
+		connection.add(SUBJECT_5, PREDICATE_1, vf.createLiteral("sfiveponecone"));
+		connection.add(SUBJECT_5, PREDICATE_1, vf.createLiteral("sfiveponectwo"), CONTEXT_2);
+		connection.add(SUBJECT_5, PREDICATE_2, vf.createLiteral("sfiveptwoctwo"), CONTEXT_2);
+		connection.commit();
+		// connection.close();
+		// connection = repository.getConnection();
+		// connection.setAutoCommit(false);
+		// test querying
+		assertQueryResult("sfourponecone", PREDICATE_1, SUBJECT_4);
+		assertQueryResult("sfourptwocone", PREDICATE_2, SUBJECT_4);
+		assertQueryResult("sfiveponecone", PREDICATE_1, SUBJECT_5);
+		assertQueryResult("sfiveponectwo", PREDICATE_1, SUBJECT_5);
+		assertQueryResult("sfiveptwoctwo", PREDICATE_2, SUBJECT_5);
+		// blind test to see if this method works:
+		assertNoQueryResult("johannesgrenzfurthner");
+		// remove a context
+		connection.clear((Resource)null);
+		connection.commit();
+		assertNoQueryResult("sfourponecone");
+		assertNoQueryResult("sfourptwocone");
+		assertNoQueryResult("sfiveponecone");
+		assertQueryResult("sfiveponectwo", PREDICATE_1, SUBJECT_5);
+		assertQueryResult("sfiveptwoctwo", PREDICATE_2, SUBJECT_5);
+	}
+
+	@Test
+	public void testFuzzyQuery()
+		throws MalformedQueryException, RepositoryException, QueryEvaluationException
+	{
+		// prepare the query
+		// search for the term "one" with 80% fuzzyness
+		StringBuilder buffer = new StringBuilder();
+		buffer.append("SELECT \n");
+		buffer.append("  Resource, Score \n");
+		buffer.append("FROM \n");
+		buffer.append("  {Resource} <" + MATCHES + "> {} ");
+		buffer.append("    <" + QUERY + "> {\"one~0.8\"}; ");
+		buffer.append("    <" + SCORE + "> {Score} ");
+		String q = buffer.toString();
+
+		// fire the query
+		TupleQuery query = connection.prepareTupleQuery(QueryLanguage.SERQL, q);
+		TupleQueryResult result = query.evaluate();
+
+		// check the results
+		BindingSet bindings = null;
+
+		// the first result is subject 1 and has a score
+		int results = 0;
+		Set<IRI> expectedSubject = new HashSet<IRI>();
+		expectedSubject.add(SUBJECT_1);
+		expectedSubject.add(SUBJECT_2);
+		expectedSubject.add(SUBJECT_3);
+		while (result.hasNext()) {
+			results++;
+			bindings = result.next();
+
+			// the resource should be among the set of expected subjects, if so,
+			// remove it from the set
+			assertTrue(expectedSubject.remove((IRI)bindings.getValue("Resource")));
+
+			// there should be a score
+			assertNotNull(bindings.getValue("Score"));
+		}
+
+		// there should have been 3 results
+		assertEquals(3, results);
+
+		result.close();
+	}
+
+	@Test
+	public void testReindexing()
+		throws Exception
+	{
+		sail.reindex();
```

Comment:
I guess the intention is to run complex query after the reindexing. Added documentation.